department-of-veterans-affairs/caseflow
# WIP | Fix TZ Problem When Switching Between Timeslots and Not - Hide Timeslots before 8:30am Local Time

### Description
Two issues were discovered pre user-testing:
1. If a users toggles "Choose a custom time" the values should stay the same. They were changing.
1. For timezones other than eastern, slot times should not display before 8:30am local time. (so, no 5:30am PDT, 6:30am PDT, 7:30am PDT slots in our pacific example).

### Testing Schedule Veteran
- Go to schedule a veteran on the "Schedule Veteran For a Hearing" page.
- Choose the Oakland RO.
- Note the timeslots are (correctly) starting at 8:30am PDT.
- Click the "Choose a custom time" link.
- Click the "Choose a timeslot" link.
- Note the timeslots are (still correctly) starting at 8:30am PDT.
- Note that there are no timeslots before 8:30am PDT
- Note that there are no timeslots after 3:30pm EST <- Note the zone
- Click one of the open timeslots, take a screenshot (cmd + shift + 4) of the form (date/time/ro are critical).
- Go look at the daily docket you scheduled this veteran on, the time should match what you scheduled.
- Do this scheduling again with a new veteran, but this time, use one of the "Custom Times" from the dropdown.
- Verify the same things, but particularly that the scheduled appointment ends up on the daily docket at the right time.

### Testing Daily Docket
- Go to the Daily Docket for the hearing you just scheduled above.
- Note that the listed time is the same as what you scheduled (be careful with the timezone label).
- On the right hand side of the screen you should see a "Time" header with three radio buttons.
- Verify that the radio buttons say:
  - "8:30am [RO LOCAL TIME FOR HEARING DAY]"
  - "11:30am [RO LOCAL TIME FOR HEARING DAY]"
  - 'Other'
- Click the 'Other' radio button, which should make a dropdown appear.
- In the dropdown, verify that the times go from "8:15 [RO LOCAL TIME FOR HEARING DAY]" to "4:45pm [RO LOCAL TIME FOR HEARING DAY]".
- Choose a time, in the dropdown, take a screenshot.
- Click "Save"
- Go back to that daily docket and note that your hearing is now at the new time.
- Try selecting one of the radio buttons and clicking save.
- Go back to that daily docket and note that your hearing is now at the new time.

### Testing Video to Virtual Conversion
- Something similar, make sure the time you schedule for is the time that appears on the daily docket.
